src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -138,6 +151,43 @@ private void throwDice(GameBean game) {
         game.setDiceThrown(true);
     }
 
+    private void buyCard(UserBean user, GameBean game, Map<String, String> returnedParams) throws PlayException, GameException {
```

Comment:
You do many actions here, that are not common for this service, I think we need something like CardUtil